mqtt/mqtt3/src/client/subscriptions.rs

```diff
@@ -251,7 +251,7 @@ impl State {
                 }
             }
 
-            let mut target_subscriptions = current_subscriptions.clone();
+            let mut target_subscriptions = std::collections::BTreeMap::new();
 
```

Comment:
why is it an empty collection? there is that second loop starting in 282 line, it checks if there is any item in the current_subscription that is no longer in the target_subscription. The original implementation copied the current_subscriptions, so if no item was removed, then the target_subscription had all items (and it generated no pending_unsubscription). Starting with an empty collection, that loop will find all existing subscription missing, and will generate pending_unsubscription for them?

Reply:
thanks! changed because it was not handling correctly the unsub